lib/kitchen/provisioner/chef_zero_scheduled_task.rb

```diff
@@ -30,19 +30,23 @@ class ChefZeroScheduledTask < ChefZero
 
       default_config :task_password
 
-      def init_command
+      def create_sandbox
+        super
         if windows_os?
-          info("Creating the scheduled task.")
-          wrap_shell_code("#{super}\n#{setup_scheduled_task_command}")
-        else
-          super
+          info("Preparing the chef-client task script")
+          prepare_client_zero_script
+          paths = config[:cookbook_files_glob].split(',') << 'chef-client-script.ps1'
+          config[:cookbook_files_glob] = paths.join(',')
         end
       end
 
       def run_command
         if windows_os?
-          script = "$script = @'\n#{scheduled_task_command_script}\n'@\n" \
-          "\n$ExecutionContext.InvokeCommand.ExpandString($Script) | out-file \"$env:temp/kitchen/chef-client-script.ps1\"" \
+          script = "#{setup_scheduled_task_command}\n" \
+          "$ScriptPath = (resolve-path $env:temp/kitchen/chef-client-script.ps1).providerpath\n" \
+          "$script = get-content -readcount 0 $ScriptPath\n" \
+          "$EnvHeader = \"`$env:Temp = '$env:temp';\"\n" \
+          "\n($EnvHeader, $Script) | out-file $ScriptPath" \
```

Comment:
I'm glad you add back the `$env:Temp` propagation, it broke the fix you did for #14